core/src/main/java/io/questdb/cairo/arr/FlatArrayView.java

```diff
@@ -177,17 +177,24 @@ default int linearSearch(double value, int offset, int length) {
 
     /**
      * Computes the average of the block of elements in this flat array.
+     * Uses Kahan summation.
      *
      * @param offset the starting offset of the block (in elements)
      * @param length the number of elements in the block
      */
     default double sumDouble(int offset, int length) {
-        //TODO: the naive summing algo doesn't compensate for accumulated error
-        double sum = 0d;
+        double sum = Double.NaN;
```

Comment:
This changes the sum initial value to NaN instead of 0. This is a bad change because when `length` is 0, the sum result is NaN, but it should be 0. This makes this implementation inconsistent with other overrides of this method.